src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/ProjectSystem/SnapshotResolver.cs

```diff
@@ -0,0 +1,108 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT license. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.IO;
+using Microsoft.AspNetCore.Razor.Utilities;
+using Microsoft.CodeAnalysis.Razor;
+using Microsoft.CodeAnalysis.Razor.ProjectSystem;
+
+namespace Microsoft.AspNetCore.Razor.LanguageServer.ProjectSystem;
+
+internal sealed class SnapshotResolver
+{
+    private readonly ProjectSnapshotManagerAccessor _projectSnapshotManagerAccessor;
+
+    // Internal for testing
+    internal readonly HostProject MiscellaneousHostProject;
+
+    public SnapshotResolver(ProjectSnapshotManagerAccessor projectSnapshotManagerAccessor)
+    {
+        _projectSnapshotManagerAccessor = projectSnapshotManagerAccessor ?? throw new ArgumentNullException(nameof(projectSnapshotManagerAccessor));
```

Comment:
Is the Accessor on the chopping block too, long term?

Reply:
I hope so, but I have to actually look at why it exists.